### DISCUSSION TITLE
Missing final version of Oziee's repo

### DISCUSSION THREAD

**Discussion by ProphetOfDoom** [BJReplay/ha-solcast-solar#17] [General]
This repo appears to have been forked from V4.0.22 of Oziees repo, so is missing his final V4.0.23 version changes, which can be found here: 

https://github.com/tabascoz/ha-solcast-solar

and are summerised as:

![image](https://github.com/BJReplay/ha-solcast-solar/assets/109991286/c8750191-7719-4698-aa1c-9fd1654c0dd8)

I don't know if they are important to anyone who adopts this repo into Home Assistant now or in the future, or whether anyone will look at merging them in in due course, but I thought I'd capture the info here for the record.

**Comment by ProphetOfDoom** [comment]
I've since seen that this subject was covered 9 days ago in this post on a thread on the HA Community Forums, so looks like you have it covered already. Well done to all, looking forward to picking up from where Oziee left off

https://community.home-assistant.io/t/solcast-global-solar-power-forecast-integration/334681/333

![image](https://github.com/BJReplay/ha-solcast-solar/assets/109991286/f85f598d-5ac0-4e68-b38f-889530308b6c)

**Reply by BJReplay** [reply]
@autoSteve has picked up some of this in a PR to be released today that picks up cached values on a 429 on start up.

Others to follow (from August when I get back from travelling, or if others are as generous as @autoSteve and contribute so much so quickly).

**Reply by autoSteve** [reply]
Oziee's 4.0.23 changes incorporated in proposed #27.

**Reply by ProphetOfDoom** [reply]
I've just zoomed forward to V4.0.29, so many thanks to all those who have helped pick up from where oziee left off.

**Comment by autoSteve** [comment]
There be a lot more in this v4.0.23 than meets the eye and the change log...

* Changing sensor class to 'total' was 'breaking' for long-term statistics. I still can't get mine back, but the historical forecast data is just 'of interest', so no 'biggie', but disappointing. I still don't know whether this should be reverted, or just lived with.
* Changing detection of config changes in ```__init__``` appears to have broken dampening automations. Yet to be confirmed.

I am going to examine the rest of the changes most carefully... Two proposed reversions so far is not cool.

**Reply by autoSteve** [reply]
Further, re. sensor class 'total'.

Historical forecasts being of interest is like looking back on what the Bureau of Met. forecast as a high temperature for December 23, 2022. Did they get it right? Ah ha! They were off by _three degrees_! Ummm. So what? We know what the actual answer was... _They_ care that they got it wrong.

So is Solcast getting more accurate, would be the question. They care. They have business customers who care. So they care, and would be tracking it. We're just interested hobbyists... and in my humble opinion just see it as missing dashed lines on a chart that should be there, yet aren't. Big deal.

So should this be a 'total' sensor really? Probably. And should my current proposed 4.0.33 reversion ever see the light of day? Probably not.

**Reply by gcoan** [reply]
It'd be preferable to have not lost the history but it's happened now so have to move forward.

If changing the sensor again loses history again then I'd prefer not to implement it

**Comment by ProphetOfDoom** [comment]
Github and Python are a bit new to me (Ada & Pl/Sql was my bag), but surely, if HA thinks it ought to be a Total, then that's what we should go with. I agree I can't see many hobbyists being very interested in what forecast generation was last month, and whether it was "correct" or not. I suspect that most will be focused on the future forecasts in order to decide when to charge batteries, charge car, turn on washing machine etc
